clash-lib/src/Clash/Core/Evaluator/Models.hs

```diff
@@ -290,7 +415,7 @@ instance AsTerm Value where
       | null bs = x
       | otherwise = Letrec bs x
      where
-      bs = Map.toList $ fmap asTerm (envLocals env)
+      bs = Map.toList . fmap asTerm $ Map.filter isRight (envLocals env)
```

Comment:
Could you add a comment why you only want the Values from the local environment? Or explain that/why the un-evaluated Terms cannot/should not be referenced by the body of the (ty)lambda.